src/atoms/fileActions.test.ts

```diff
@@ -0,0 +1,329 @@
+    });
+
+    expect(leftPane.current.activeFile).not.toEqual(fileC);
+    expect(leftPane.current.activeFile).toEqual(fileB);
+  });
+
+  test('should not make another file active, in same pane, after closing other than the active', () => {
+    const store = createStore();
+    const { result: openFile } = renderHook(() => useSetAtom(openFileAtom, { store }));
+    const { result: closeFileFromPane } = renderHook(() => useSetAtom(closeFileFromPaneAtom, { store }));
+    const { result: leftPane } = renderHook(() => useAtomValue(leftPaneAtom, { store }));
+
+    const fileA = makeFile('fileA.txt');
+    const fileB = makeFile('fileB.txt');
+    const fileC = makeFile('fileC.txt');
+
+    act(() => {
+      openFile.current(fileA);
+      openFile.current(fileB);
+      openFile.current(fileC);
+    });
+
+    expect(leftPane.current.activeFile).toEqual(fileC);
+
+    act(() => {
+      closeFileFromPane.current({ paneId: leftPane.current.id, fileId: fileB.path });
+    });
+
+    expect(leftPane.current.activeFile).toEqual(fileC);
+  });
+
+  test('should not open folder entries', () => {
+    const store = createStore();
+    const { result: openFileResult } = renderHook(() => useSetAtom(openFileAtom, { store }));
+    const { result: leftPane } = renderHook(() => useAtomValue(leftPaneAtom, { store }));
+    const { result: rightPane } = renderHook(() => useAtomValue(rightPaneAtom, { store }));
+
+    expect(leftPane.current.files.length).toBe(0);
+    expect(rightPane.current.files.length).toBe(0);
+
+    act(() => openFileResult.current(makeFolder('uploads')));
+
+    expect(leftPane.current.files.length).toBe(0);
+    expect(rightPane.current.files.length).toBe(0);
+  });
+
+  test('should focus panel to make active', () => {
+    const store = createStore();
+    const { result: activePane } = renderHook(() => useAtomValue(activePaneAtom, { store }));
+    const { result: focusPanel } = renderHook(() => useSetAtom(focusPaneAtom, { store }));
+
+    const paneToFocus: PaneId = activePane.current.id === 'LEFT' ? 'RIGHT' : 'LEFT';
+
+    act(() => {
+      focusPanel.current(paneToFocus);
+    });
+
+    expect(activePane.current.id).toBe(paneToFocus);
+  });
+
+  test('should read file content on openFile', async () => {
+    const store = createStore();
+    const { result: openFile } = renderHook(() => useSetAtom(openFileAtom, { store }));
+    const { result: leftPane } = renderHook(() => useAtomValue(leftPaneAtom, { store }));
+
+    const mockedReadFileContent = vi.mocked(readFileContent).mockResolvedValue({
+      type: 'tiptap',
+      textContent: 'some content',
+    });
+
+    const fileA = makeFile('fileA.txt');
+
+    act(() => {
+      openFile.current(fileA);
+    });
+
+    expect(leftPane.current.activeFileContent).toBeDefined();
+    const { result: activeFile } = renderHook(() => useAtomValue(leftPane.current.activeFileContent!, { store }));
```

Comment:
The render syntax is a bit heavy to me. Maybe it would make sense to create utils function `renderSetAtomHook`, `renderGetAtomHook` and `renderAtomHook` that would take an atom and a store and return `result` directly. What do you think?